### ISSUE TITLE
Routes inheriting parent routes

### ISSUE THREAD

**Issue by tskweres** [mtrpcic/pathjs#54]
I'm having trouble with routes that are deeper than 2 layers.

For example, i want

site.com/users

site.com/users/friends

site.com/users/friends/friend-3

Is it possible to have users/friends inherit the enter / exit functions from user?

Any ideas?

**Comment by tommycutter** [comment]
Did you ever find a solution to this? I see that you can include "optional" parameters, but ideally I'd like to keep page routes separate.

**Comment by tskweres** [comment]
I had to build my own using HTML5 data attributes. 

I had 4 layers of pages, data-page="1", data-page="2" and so on. 

And then I had data-parents, so if a page was showing, I checked to see if it had a data-parent, and if it did, I showed that page as well. 

My entire app is hide/show in one file, that's what presented the challenge.

**Comment by tommycutter** [comment]
Ah, sweet. After some experimentation, I ended up doing something like this, which works well for most dynamic parts of my app (more robust routing would be ideal though):

https://gist.github.com/tommycutter/5154572

**Comment by tskweres** [comment]
Yeah I've looked at a lot of js frameworks I'm surprised child and
grandchild routes and route inheritance aren't a bigger problem for people.
Am I the only one with more than 1 view on my app? :P
